craft_providers/lxd/launcher.py

```diff
@@ -551,6 +552,80 @@ def _set_timezone(instance: LXDInstance, project: str, remote: str, lxc: LXC) ->
     )
 
 
+def _wait_for_instance_ready(instance: LXDInstance) -> None:
+    """Wait for an instance to be ready.
+
+    If another process created an instance and is still running, then wait for
+    the other process to finish setting up the instance.
+
+    :param instance: LXD instance to wait for.
+
+    :raises LXDError:
+    - If the instance is not ready and the process that created the
+    instance is no longer running.
+    - If the function times out while waiting for another process
+    to set up the instance.
+    - On any failure to get config data or info from the instance.
+    """
+    instance_state = instance.info().get("Status")
+    instance_status = instance.config_get("user.craft_providers.status")
+
+    # check if the instance is ready
+    if (
+        instance_status == ProviderInstanceStatus.FINISHED.value
+        and instance_state == "STOPPED"
+    ):
+        logger.debug("Instance %r is ready.", instance.instance_name)
+        return
+
+    logger.debug(
+        "Instance %r is not ready (State: %s, Status: %s)",
+        instance.instance_name,
+        instance_state,
+        instance_status,
+    )
+
+    # LXD is linux-only, but verify the platform anyways
+    if sys.platform == "linux":
+        # get the PID of the process that created the instance
+        pid = instance.config_get("user.craft_providers.pid")
+
+        # an empty string means there was no value set
+        if pid == "":
+            raise LXDError(
+                brief="Instance is not ready.",
+                details=(
+                    f"Instance {instance.instance_name!r} is not ready and does not "
+                    "have the pid of the process that created the instance."
+                ),
+            )
+
+        # check if the PID is active
+        if Path(f"/proc/{pid}").exists():
+            logger.debug(
+                "Process that created the instance %r is still running (pid %s).",
+                instance.instance_name,
+                pid,
+            )
```

Comment:
You may also want to check if `/proc/{pid}/exe` is symlink to a python.

Reply:
If we have permission to do so. For a process run by another user you won't be able to see the destination of the symlink unless you're root. (Easy test: set `pid` to 1, which should always be owned by root)

Although you can mostly get around that by reading the first string in `/proc/{pid}/cmdline`

Reply:
I documented this enhancement here: https://github.com/canonical/craft-providers/issues/468